### ISSUE TITLE
Elasticsearch does not install service correctly in CentOS (EnvironmentFile configuration setting improperly set)

### ISSUE THREAD

**Issue by pgibler** [elasticsearch/puppet-elasticsearch#192]
I noticed that my elasticsearch wasn't loading. Right now, it appears to me that the EnvironmentFile value that gets inserted does not properly get the $name value. But that's just a guess - I haven't been able to figure out the issue on its own, as it appears the library was correctly written to add the $name value into the EnvironmentFile setting. Any help would be massively appreciated.

I'm initializing elasticsearch in my puppet manifest as as so. 

```
# Requires Puppet 3.4.6+ to run

# --- Configuration options ---------------------------------------------------

# Set the right versions for our installation.
$elasticsearch_application_version = '1.3'
$elasticsearch_ec2_plugin_version = '2.3.0'

# Load the appname_config module.
$module_path = get_module_path('appname_config')

# Gather the configuration options for elasticsearch.yml.
$config_options = loadyaml("${module_path}/files/elasticsearch.yml")

# Gather the logging options for logging.yml.
$logging_options = loadyaml("${module_path}/files/logging.yml")

# --- Allow virtual -----------------------------------------------------------

Package {
allow_virtual => true,
}

# --- Files & Folders ---------------------------------------------------------

# Stores elasticsearch folders.
file { '/mnt/elasticsearch':
  ensure => directory
}

# Stores elasticsearch logs.
file { '/mnt/elasticsearch/logs':
  ensure => directory
}

# Stores elasticsearch data.
file { '/mnt/elasticsearch/data':
  ensure => directory
}

# --- Packages ----------------------------------------------------------------

# Used for interacting with the elasticsearch HTTP API.
package { 'curl':
  ensure => installed
}

# Java is a dependency of Elasticsearch.
# Use 'default-jdk' for Debians-based distros.
package { 'java-1.7.0-openjdk-devel':
  ensure => installed
}

# --- Yum repo ----------------------------------------------------------------

yumrepo { 'elasticsearch':
  name => 'elasticsearch',
  descr => "Elasticsearch repository for ${elasticsearch_application_version} packages",
  baseurl => "http://packages.elasticsearch.org/elasticsearch/${$elasticsearch_application_version}/centos",
  gpgcheck => 1,
  gpgkey => 'http://packages.elasticsearch.org/GPG-KEY-elasticsearch',
  enabled => 1
}

# --- Elasticsearch------------------------------------------------------------

# appname preferred full-text search technology.
class { 'elasticsearch':
  require => [ Yumrepo['elasticsearch'] ]
}

# Set the instance name to elasticsearch-appname.
elasticsearch::instance { 'appname':
  config => $config_options,
  init_defaults => {
  'ES_CLUSTER_NAME' => $es_cluster_name
  }
}

# Install aws cluster plugin for auto-sync
elasticsearch::plugin{"elasticsearch/elasticsearch-cloud-aws/${elasticsearch_ec2_plugin_version}":
  module_dir => 'es-cloud-aws',
  instances  => 'appname'
}
```

After I run puppet, I'm notified that the service was installed successfully. Obviously, this ends up not being true.

```
==> elasticsearch: Notice: /Stage[main]/Main/Elasticsearch::Instance[appname]/Elasticsearch::Service[appname]/Elasticsearch::Service::Init[appname]/Augeas[defaults_appname]/returns: executed successfully
==> elasticsearch: Notice: /Stage[main]/Main/Elasticsearch::Instance[appname]/Elasticsearch::Service[appname]/Elasticsearch::Service::Init[appname]/File[/etc/init.d/elasticsearch-appname]/ensure: created
==> elasticsearch: Notice: /Stage[main]/Main/Elasticsearch::Instance[appname]/Elasticsearch::Service[appname]/Elasticsearch::Service::Init[appname]/Service[appname]/ensure: ensure changed 'stopped' to 'running'
==> elasticsearch: Notice: Finished catalog run in 109.69 seconds
```

The error occurs as I try to set a configuration variable for the elasticsearch service, which gets written to what seems to be the right EnvironmentFile (elasticsearch-appname) correctly, but the service itself points to a non-existent EnvironmentFile - the suffix `-appname` does not get included in the EnvironmentFile setting.

```
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: - IllegalArgumentException[Could not resolve placeholder 'ES_CLUSTER_NAME']
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: java.lang.IllegalArgumentException: Could not resolve placeholder 'ES_CLUSTER_NAME'
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: at org.elasticsearch.common.property.PropertyPlaceholder.parseStringValue(PropertyPlaceholder.java:124)
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: at org.elasticsearch.common.property.PropertyPlaceholder.replacePlaceholders(PropertyPlaceholder.java:81)
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: at org.elasticsearch.common.settings.ImmutableSettings$Builder.replacePropertyPlaceholders(ImmutableS...ava:1060)
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: at org.elasticsearch.node.internal.InternalSettingsPreparer.prepareSettings(InternalSettingsPreparer.java:101)
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: at org.elasticsearch.bootstrap.Bootstrap.initialSettings(Bootstrap.java:106)
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: at org.elasticsearch.bootstrap.Bootstrap.main(Bootstrap.java:177)
Oct 10 11:40:50 appname-elasticsearch-box elasticsearch-appname[11482]: at org.elasticsearch.bootstrap.Elasticsearch.main(Elasticsearch.java:32)
Oct 13 05:45:38 appname-elasticsearch-box systemd[1]: Started LSB: This service manages the elasticsearch daemon.
```

The service file is as follows:

```
[vagrant@appname-elasticsearch-box ~]$ cat /usr/lib/systemd/system/elasticsearch.service
[Unit]
Description=Starts and stops a single elasticsearch instance on this system
Documentation=http://www.elasticsearch.org

[Service]
Type=forking
EnvironmentFile=/etc/sysconfig/elasticsearch
User=elasticsearch
Group=elasticsearch
PIDFile=/var/run/elasticsearch/elasticsearch.pid
ExecStart=/usr/share/elasticsearch/bin/elasticsearch -d -p /var/run/elasticsearch/elasticsearch.pid -Des.default.config=$CONF_FILE -Des.default.path.home=$ES_HOME -Des.default.path.logs=$LOG_DIR -Des.default.path.data=$DATA_DIR -Des.default.path.work=$WORK_DIR -Des.default.path.conf=$CONF_DIR
# See MAX_OPEN_FILES in sysconfig
LimitNOFILE=65535
# See MAX_LOCKED_MEMORY in sysconfig, use "infinity" when MAX_LOCKED_MEMORY=unlimited and using bootstrap.mlockall: true
#LimitMEMLOCK=infinity
# Shutdown delay in seconds, before process is tried to be killed with KILL (if configured)
TimeoutStopSec=20

[Install]
WantedBy=multi-user.target
```

The variables file is as follows:

```
[vagrant@appname-elasticsearch-box ~]$ cat /etc/sysconfig/elasticsearch-appname                                                                                                          CONF_DIR=/etc/elasticsearch/appname
CONF_FILE=/etc/elasticsearch/appname/elasticsearch.yml
ES_CLUSTER_NAME=vagrant-cluster
ES_GROUP=elasticsearch
ES_HOME=/usr/share/elasticsearch
ES_USER=elasticsearch
LOG_DIR=/var/log/elasticsearch/appname
```

Notice how the EnvironmentFile points to `/etc/sysconfig/elasticsearch` and not `/etc/sysconfig/elasticsearch-appname`.

Any idea why this is happening?

**Comment by electrical** [comment]
Just a heads up. Im now working on solving this issue and making it support centos7 better.

**Comment by electrical** [comment]
Hi,

Centos7 Support ( mainly systemd support ) has been improved via https://github.com/elasticsearch/puppet-elasticsearch/pull/258

**Comment by electrical** [comment]
I'm closing this issue.
SystemD improvements have been released with version 0.9.0
